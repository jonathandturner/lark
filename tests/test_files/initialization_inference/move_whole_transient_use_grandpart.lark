//~ execute:no

struct Foo { x: uint }
struct Bar { foo: Foo }

def main() {
  let bar = Bar(foo: Foo(x: 22))
  take_bar(bar)
  bar.foo.x = 44
  //~ ERROR: access to uninitialized path
}

def take_bar(v: Bar) { }
