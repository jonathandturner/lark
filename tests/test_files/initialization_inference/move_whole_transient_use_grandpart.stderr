error: access to uninitialized path
- initialization_inference/move_whole_transient_use_grandpart:9:2
9 |   bar.foo.x = 44
  |   ^^^^^^^^^^^^^^
